typify-impl/src/convert.rs

```diff
@@ -1011,14 +1011,14 @@ impl TypeSpace {
         ];
 
         if let Some(format) = format {
-            if let Some((_, ty, imin, imax)) = formats
+            if let Some((_fmt, ty, imin, imax)) = formats
                 .iter()
                 .find(|(int_format, _, _, _)| int_format == format)
             {
                 // If the type matches with other constraints, we're done.
                 if multiple.is_none()
-                    && (min.is_none() || min == Some(*imin))
-                    && (max.is_none() || max == Some(*imax))
+                    && (min.map(|f| f.ge(imin)).unwrap_or(false))
+                    && (max.map(|f| f.le(imax)).unwrap_or(false))
```

Comment:
this doesn't seem quite right. Previously the logic said: "if it's None OR is the specified bound". Now this says "if it's Some AND is within the specified bound". So this will only apply if both the `minimum` and `maximum` values are set in the schema.

I'm not sure of the best path forward here. I tried adjusting this, but then the `NonZeroUXX` tests failed. One option might be to specify add formats for the `NonZeroXX` types in the table above and then here to `filter` rather than `find`.